worker/caasapplicationprovisioner/application.go

```diff
@@ -210,32 +250,58 @@ func (a *appWorker) loop() error {
 		return nil
 	}
 
+	var scaleChan <-chan time.Time
+	var scaleTries int
+	var trustChan <-chan time.Time
+	var trustTries int
+	const maxRetries = 20
+	const retryDelay = 3 * time.Second
+
 	for {
 		select {
 		case _, ok := <-appScaleWatcher.Changes():
 			if !ok {
 				return fmt.Errorf("application %q scale watcher closed channel", a.name)
 			}
-			if err := a.ensureScale(app); err != nil {
-				return err
+			scaleTries = 0
+			scaleChan = time.After(0)
+		case <-scaleChan:
+			err := a.ensureScale(app)
+			if errors.IsNotFound(err) {
+				if scaleTries >= maxRetries {
+					return errors.Annotatef(err, "more than %d retries ensuring scale", maxRetries)
+				}
+				scaleTries++
+				scaleChan = time.After(retryDelay)
+			} else if err != nil {
+				return errors.Trace(err)
 			}
 		case _, ok := <-appTrustWatcher.Changes():
 			if !ok {
 				return fmt.Errorf("application %q trust watcher closed channel", a.name)
 			}
-			if err := a.ensureTrust(app); err != nil {
-				return err
+			trustTries = 0
+			trustChan = time.After(0)
```

Comment:
if trust chan is not nil, no need to schedule a new one.

Reply:
Done